Add Action Push Native delivery method

excid3 opened the pull request:
This adds support for [Action Push Native](https://github.com/rails/action_push_native) as a delivery method.

```ruby
class CommentNotifier < ApplicationNotifier
  deliver_by :action_push_native do |config|
    config.devices = -> { ApplicationPushDevice.where(owner: recipient) }
    config.format = -> {
      {
        title: "Hello world, #{recipient.first_name}!",
        body: "Welcome to Noticed with Action Push Native.",
        badge: 1,
      }
    }
    config.apple_data = -> {
      { category: "observable" }
    }
    config.google_data = -> {
      { }
    }
    config.with_data = -> {
      { }
    }
  end
end
```

It also adds a deprecation notice to iOS and FCM delivery methods which are replaced by Action Web Push.

jcmaciel commented:
@excid3 hey, when merge it?

jespr commented:
@excid3 in your example in the PR and in `active_push_native.md` you have `config.apple_data {}` and `config.google_data {}` but it looks like your code expects `config.apple {}` and `config.google {}`? :)

jespr commented on lib/noticed/delivery_methods/action_push_native.rb:
```diff
@@ -0,0 +1,23 @@
+          .with_apple(evaluate_option(:apple))
+          .with_google(evaluate_option(:google))
+          .with_data(evaluate_option(:data))
+          .new(evaluate_option(:format))
```
should this be `.new(**evaluation_option(:format))` - I think I run into an error if not

excid3 commented:
@jespr just pushed some changes for that:

- fixed the arguments for `new`
- fixed the mismatched config names
- renamed them to match the factory methods on APN since that's exactly what they're calling
- updated examples

djpremier commented on lib/noticed/delivery_methods/action_push_native.rb:
```diff
@@ -0,0 +1,24 @@
+      required_options :devices, :format
+
+      def deliver
+        notification = evaluate_option(:silent) ? notification_class : notification_class.silent
```
https://github.com/excid3/noticed/pull/560
